patches/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -122,3 +138,33 @@
                          }
      
                          @Override
+@@ -1745,7 +_,7 @@
+             throw new IllegalStateException("Client acknowledged config, but none was requested");
+         } else {
+             this.connection
+-                .setListener(new ServerConfigurationPacketListenerImpl(this.server, this.connection, this.createCookie(this.player.clientInformation())));
++                .setListener(new ServerConfigurationPacketListenerImpl(this.server, this.connection, this.createCookie(this.player.clientInformation(), this.isModdedConnection)));
+         }
+     }
+ 
+@@ -1779,5 +_,20 @@
+     @FunctionalInterface
+     interface EntityInteraction {
+         InteractionResult run(ServerPlayer p_143695_, Entity p_143696_, InteractionHand p_143697_);
++    }
++    
++    @Override
++    public void handleCustomPayload(net.minecraft.network.protocol.common.ServerboundCustomPayloadPacket p_294276_) {
++        if (!this.isModdedConnection) {
++            super.handleCustomPayload(p_294276_);
++        }
++        
++        net.neoforged.neoforge.network.registration.NetworkRegistry.getInstance().onModdedPacketAtServer(
++                this, p_294276_
++        );
++    }
++
++    public ServerGamePacketListener self() {
```

Comment:
Why is this done here rather than how we do other extension interfaces of having the `self` method be private and just cast to the expected type?

Reply:
Because here ther eis a method conflict and its needs to be overriden.